src/PhpParser/Comparing/ConditionSearcher.php

```diff
@@ -67,6 +79,15 @@ private function searchElseForVariableRedeclaration(Assign $assign, Else_ $else)
                 continue;
             }
 
+            $isFoundInExpr = (bool) $this->betterNodeFinder->findFirst(
```

Comment:
I believe this could lead to a false negative on this code:
```php
$text = 'a';
if ($flag) {
    $text = 'a';
    $text = $text . "b";
}
else {
    $text = 'a';
    $text = $text . "c";
}
```
The variable is used in an assignment but not in the first re-assignment.